File: backend/src/main/java/com/online/giftshop/services/DeliveryHistoryService.java
/**
 * @author Gagandeep Singh
 * @email singh.gagandeep3911@gmail.com
 * @create date 2021-01-28 00:32:22
 * @modify date 2021-01-28 00:32:22
 * @desc [description]
 */
package com.online.giftshop.services;

import java.util.List;

import com.online.giftshop.entities.DeliveryHistory;
import com.online.giftshop.enums.OrderStatus;

public interface DeliveryHistoryService {

  List<DeliveryHistory> fetchDeliveriesByOrderId(Long orderId);

  DeliveryHistory createEntry(Long orderId, OrderStatus status);
  
}

File: frontend/src/app/services/delivery-history.service.ts
/**
 * @author Gagandeep Singh
 * @email singh.gagandeep3911@gmail.com
 * @create date 2021-01-26 11:23:10
 * @modify date 2021-01-26 11:23:10
 * @desc [description]
 */
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root',
})
export class DeliveryHistoryService {
  orderServiceUrl = `${environment.protocol}${environment.applicationUrl}/delivery`;

  constructor(private http: HttpClient) {}

  fetchDeliveryByOrderId(orderId) {
    return this.http.get(`${this.orderServiceUrl}/${orderId}`);
  }
}
